fastparquet/writer.py

```diff
@@ -1349,7 +1347,7 @@ def partition_on_columns(data, columns, root_path, partname, fmd,
     """
     # Pandas groupby has by default 'sort=True' meaning groups are sorted
     # between them on key.
-    gb = data.groupby(columns if len(columns) > 1 else columns[0])
+    gb = data.groupby(columns if len(columns) > 1 else columns[0], observed=False)
```

Comment:
It would be nice to have a comment here to explain why we need `observed=False`.

Reply:
This would need a more thorough investigation than I did, I was only interested in keeping behaviour consistent and silencing the deprecation warnings.